libs/ui-components/src/components/Device/DevicesPage/DeviceToolbarFilters.tsx

```diff
@@ -137,10 +137,13 @@ const LabelFleetSelector = ({ selectedFleetNames, selectedLabels, onSelect, plac
   };
 
   const fetchTextMatches = async (val: string) => {
-    // Matches for labels are always exact matches
-    const labelMatches = get<DeviceList>(`devices?labelSelector=${val}&limit=1&sortBy=metadata.name&sortOrder=Asc`);
+    // By default (without the "equal" sign), the API returns a partial match, but only on the label keys
+    // To prevent this confusing behaviour, we query for exact matches in all cases (for both keys and values)
+    const exactLabelMatchParam = val.includes('=') ? val : `${val}=`;
+    const labelMatches = get<DeviceList>(
+      `devices?labelSelector=${exactLabelMatchParam}&sortBy=metadata.name&sortOrder=Asc`,
```

Comment:
not for this PR, but shoudn't we build these `*Matches` strings using the `URLSearchParams` ?

Reply:
Yes, that's a good point.